File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.js
function solution(land) {
    //top->down
    let N=land.length
    for(let i=1;i<N;i++){
        for(let j=0;j<4;j++){
            land[i][j]+=Math.max(...land[i-1].slice(0,j),...land[i-1].slice(j+1))
        }
    }
    return Math.max(...land[N-1])
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    N=len(land)
    for i in range(N-1,0,-1):
        temp_list=land[i]
        for j in range(4):
            temp_list=land[i].copy()
            temp_list.pop(j)
            land[i-1][j]+=max(temp_list)

    return max(land[0])